### PR TITLE
Hari

### PR DESCRIPTION
**Pull request by Hariprasadm1408**
Adding Unit testing functions

### CODE REVIEW

**Review by divyalakshmi-padmanaban — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+│   ├── add_passenger_floor.h
+│   ├── create_floor_array.h
+│   ├── add_passenger_floor.h
+│   ├── remove_passenger_floor.h
+│   ├── remove_passenger_queue.h
+│   ├── emergency_stop _handling.h
+│   ├── average_maximum waiting_time.h
+│   ├── output_log_file.h
+│   └── cli_update.h
+├── install.sh 	-> Installation script
```
Comment (MaazJamal) [inline README.md:59]: Do we have an install script?

File: doc/Developer_Manual.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Elevator simulation is a software to simulate the operations of multiple elevators in a commercial building, by creating an algorithm.
+
+# Requirement
+
+Software simulates Elevator scheduling in a building for an day by reading the input file with Paaseger ID,timestamp,start and destination floor.
+
+# Assumption
+
+- User gives the number of floors and elevators in the building, the value should be less than 100.
```
Comment (MaazJamal) [inline doc/Developer_Manual.md:11]: less than or equal to 100

File: doc/Developer_Manual.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  - The function in moving_lift_down.c to move the lift down.
+  - The function in moving_lift_up.cc to move the lift up.
+  - The function in average_maximum_waiting_time.c to calculate the average and maximum waiting time.
+  - The function in is_lift_empty.c to check the space in elevator
+  - The function in remove_passenger_floor.c to remove the passenger from the floor array.
+The dynamc_simulation function calls:
+  - The function in input_static_dynamic.c to get version input from user.
+  - The function in read_user_input_validate.c to validate the inputs from user console.
+  - The function in emergency_stop _handling.c to to bring the lifts to 1st floor incase of emergency
+  - The function in output_log_file.c to write the output files
```
Comment (MaazJamal) [inline doc/Developer_Manual.md:48]: This has been changed into travel_log_file

File: INSTALL.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Installation
+
+## Supported Platforms:
+
+Unix, Cygwin and MacOS.
+
+## Prerequisite
+
+GCC Compiler, GNU make.
+If using Cygwin, dos2unix will be required
```
Comment (MaazJamal) [inline INSTALL.md:10]: Do we need this doc2unix?

File: INSTALL.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+## No Dependent External Libraries
+
+## Build Project:
+
+```
+make
+```
+
+Excutable binary Elevator_simulator file will be placed at `bin` directory.
```
Comment (MaazJamal) [inline INSTALL.md:20]: executable binary will be in root folder and not in 'bin' directory

File: README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+│   ├── create_elevator_array.c
+│   ├── add_request_queue.c
+│   ├── add_passenger_floor.c
+│   ├── create_floor_array.c
+│   ├── add_passenger_floor.c
+│   ├── remove_passenger_floor.c
+│   ├── remove_passenger_queue.c
+│   ├── emergency_stop _handling.c
+│   ├── average_maximum waiting_time.c
+│   ├── output_log_file.c
```
Comment (MaazJamal) [inline README.md:85]: changed to travel_log_file.c

File: doc/Developer_Manual.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  - The function in cli_update.c to display elevator information on screen.
+  - The function in input_after_stop.c to get input from user when emergency situation encountered. 
+
+# Documentation
+
+Complete documentation of the source code can be found [here](https://github.com/MaazJamal/elevator_simulator/blob/devel/doc/html/index.html).
```
Comment (MaazJamal) [inline doc/Developer_Manual.md:54]: i dont think we have /blob/ in our address

File: doc/User_Manual.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
+# Introduction
+
+Elevator_Simulation is a software that operates any number of elevator in a commercial building with given number of floors.
+
+# Installation
+
+Please refer to [INSTALL.md](https://github.com/MaazJamal/elevator_simulator/blob/devel/INSTALL.md) file for installation instructions.
```
Comment (MaazJamal) [inline doc/User_Manual.md:7]: is this address valid?

File: README.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
+# Elevator_Simulator
+
+Elevator_Simulator is a software that simulates Elevator in a commercial building.
+
+
+# Authors
+
+- Hariprasad Munusamy - HariprasadMunusamy@cmail.carleton.ca (101132017) - Carleton University
```
Comment (MaazJamal) [inline README.md:8]: is it nessary to put out emails and ids?  Its information we shouldnt really share out in public.

File: README.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+- Maaz Jamal  - maazjamal@cmail.carleton.ca (101127351) - Carleton University
+- Ritika Arora - ritikaarora@cmail.carleton.ca (101141126) - Carleton University
+- Divyalakshmi Padmanaban  - Dpadm049@uottawa.ca (300110144) - Carleton University
+
+# Installation
+
+Installation steps are mentioned in [INSTALL.md](https://github.com/MaazJamal/elevator_simulator/tree/master/INSTALL.md)
+
```
Comment (MaazJamal) [inline README.md:16]: mention local path so just Install.md

File: test/test_cases.c [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+	printf("4.To test add_request_queue\n");
+	printf("5.To test input_lifts\n");
+	printf("6.To test moving_lift_down\n");
+	printf("7.To test passenger_take_in\n");
+	printf("8.To test remove_passenger_queue\n");
+	scanf("%d",&test_choice);
+	//switch to execute
+	switch(test_choice)
+	{
+		case 1: test_count_passenger();
```
Comment (MaazJamal) [inline test/test_cases.c:20]: Not every test produces and output text file.

File: src/main.c [hunk 12]
```diff
@@ -1,13 +1,71 @@
...
+                		/* Run automated unit testing */
+                		printf("Running automated unit testing\n");
+                		test_cases();
+                		printf("Automated testing has completed\n");
+                		return 0;
+               			break;
+            		case 'e':
+                		/* Main Elevator simulation program */
+                		printf("Running Elevator Simulation Program\n");
+                		int elevator_version=input_static_dynamic();
```
Comment (MaazJamal) [inline src/main.c:51]: This is for release 2. We are currently on release 1

**Pull request merged**